src/main/java/com/github/davidmoten/rx/jdbc/RegexMappableResultSet.java

```diff
@@ -0,0 +1,830 @@
+package com.github.davidmoten.rx.jdbc;
+
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.*;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class RegexMappableResultSet implements ResultSet {
+
+    private final ResultSet rs;
+    private final int columnCount;
+    private final String[] columnLabels;
+    private final HashMap<String,Integer> cachedIndices = new HashMap<>(); //-1 means unfound
+
+    private RegexMappableResultSet(ResultSet rs) {
```

Comment:
Happy to have a private constructor but I can't see a static factory method to create one of these